app/controllers/users/omniauth_callbacks_controller.rb

```diff
@@ -0,0 +1,13 @@
+class Users::OmniauthCallbacksController < Devise::OmniauthCallbacksController
+  def google_oauth2
+      @user = User.from_omniauth(request.env["omniauth.auth"])
+
+      if @user.persisted?
```

Comment:
Here's a two point plan for fixing this: 

To prevent bad sign ins from erroring out:
```
# in this controller
  rescue_from Mongoid::Errors::DocumentNotFound,
              with: -> { redirect_to root_path }

# in user model
  def self.from_omniauth(access_token)
    data = access_token.info
    user = User.find_by email: data['email'] # this replaces the `where`. A little cleaner since we're unique on email

    user
  end

```